### PR TITLE
Split the oneof Enum generation.

### PR DESCRIPTION
**Pull request by thomasvl**
Move the fileprivate impl details down so the enum defintion is shorter
and more likely just want someone reading the source needs.

### CODE REVIEW

**Comment by thomasvl** [comment]
This leaves the `Equatable` conformance and `==` generation on the main Enum.  We could move this down, but it also sorta seems like something that we might want to keep up top so it is more discoverable, thoughts?

File: Sources/protoc-gen-swift/MessageGenerator.swift [hunk 1]
```diff
@@ -327,6 +327,10 @@ class MessageGenerator {
     // Nested messages
     //
 
+    for o in oneofs {
+      o.generateRuntimeSupport(printer: &p)
```
Comment (tbkka) [inline Sources/protoc-gen-swift/MessageGenerator.swift]: Just a thought:  `generateExtensions` might be a more descriptive name here.

Comment (thomasvl) [inline Sources/protoc-gen-swift/MessageGenerator.swift]: _Extensions_ is always an overloaded terms in protos and Swift…  If we have a better name let's bulk do it after all these PRs land.

**Comment by thomasvl** [comment]
@tbkka any thoughts on https://github.com/apple/swift-protobuf/pull/454#issuecomment-291546088 (Equatable and ==)?

File: Sources/protoc-gen-swift/MessageGenerator.swift (hunk 1 above)
Comment (tbkka) [inline Sources/protoc-gen-swift/MessageGenerator.swift]: Good point.  😛

**Comment by tbkka** [comment]
I think I agree with leaving it where it is.  The generated code for `==` is pretty straightforward, so doesn't provide much in the way of visual clutter.

File: Sources/protoc-gen-swift/MessageGenerator.swift (hunk 1 above)
Comment (thomasvl) [inline Sources/protoc-gen-swift/MessageGenerator.swift]: generateRuntimeExtensions?  generateSwiftExtensions?  generateRuntimeExtras?

**Pull request merged**